sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,201 @@
+from sympy import Symbol
+from sympy.physics.vector import Vector, cross
+from sympy.physics.mechanics import (dynamicsymbols, Body, PinJoint,
+                                     SlidingJoint, CylindricalJoint, PlanarJoint,
+                                     SphericalJoint)
+
+
+def test_pin_joint():
+    parent = Body('parent')
+    child = Body('child')
+    l = Symbol('l')
+    pin_joint = PinJoint('pin_joint', parent, child, child_point_pos=(0, l, 0))
+
+    assert pin_joint.name == 'pin_joint'
+    assert pin_joint.parent == parent
+    assert pin_joint.child == child
+    assert pin_joint.parent_joint_vector == Vector(0)
+    assert pin_joint.child_joint_vector == l * child.frame.y
+    assert pin_joint.parent_axis == parent.frame.x
+    assert pin_joint.child_axis == child.frame.x
+
+    theta = dynamicsymbols('pin_joint_theta')
+    thetad = dynamicsymbols('pin_joint_theta', 1)
+    omega = dynamicsymbols('pin_joint_omega')
+    assert pin_joint.coordinates == [theta]
+    assert pin_joint.speeds == [omega]
+    assert pin_joint.kds == [thetad - omega]
+
+    child_joint_point = pin_joint.child_joint_point
+    parent_joint_point = pin_joint.parent_joint_point
+    assert parent_joint_point.name == 'pin_joint_parent_joint'
+    assert child_joint_point.name == 'pin_joint_child_joint'
+    assert child_joint_point.pos_from(parent_joint_point) == Vector(0)
+    assert child_joint_point.pos_from(child.masscenter) == l * child.frame.y
+    assert parent_joint_point.pos_from(parent.masscenter) == Vector(0)
```

Comment:
You are missing the most important one, `child_masscenter.pos_from(parent_masscenter)`.